concepts/70 Data Binding/51 Data Source Examples/3 Custom Sources/0 Connect to RESTful Service.md

```diff
@@ -10,43 +10,257 @@ Such services can have their own URL conventions and additional query-string par
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            });
         },
 
-        update: function(key, values) {
-            return $.ajax({
-                url: SERVICE_URL + "/" + encodeURIComponent(key),
+        update: (key, values) => {
+            return fetch(SERVICE_URL + "/" + encodeURIComponent(key), {
                 method: "PUT",
-                data: values
+                body: JSON.stringify(values),
+                headers: {
+                    'Content-Type': 'application/json'
+                }
             });
         },
 
-        remove: function(key) {
-            return $.ajax({
-                url: SERVICE_URL + "/" + encodeURIComponent(key),
+        remove: (key) => {
+            return fetch(SERVICE_URL + "/" + encodeURIComponent(key), {
                 method: "DELETE",
             });
+        },
+    });
+
+    export default {
+        components: {
+            // ...
+        },
+        data() {
+            return {
+                customDataSource
+            }
         }
+    }
+    </script>
 
-    });
+    <!-- tab: App.vue (Composition API) -->
+    <template>
+        <!-- ... -->
+    </template>
+
+    <script setup>
+    // ...
+    import CustomStore from 'devextreme/data/custom_store';
+    import 'whatwg-fetch';
     
-Note that all user functions return the result of the jQuery AJAX call, which is compatible with the **jQuery.Deferred** promise. In fact, you may use any promise-compatible object to connect to any asynchronous data storage; for example - to an HTML5 File API, and not necessarily to HTTP endpoints.
+    const customDataSource = new CustomStore({
+        
+        load: (loadOptions) => {
+            return fetch(SERVICE_URL);
+        },
+
+        byKey: (key) => {
+            return fetch(SERVICE_URL + "/" + encodeURIComponent(key));
+        },
+
+        insert: (values) => {
+            return fetch(SERVICE_URL, {
+                method: "POST",
+                body: JSON.stringify(values),
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            });
+        },
+
+        update: (key, values) => {
+            return fetch(SERVICE_URL + "/" + encodeURIComponent(key), {
+                method: "PUT",
+                body: JSON.stringify(values),
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            });
+        },
+
+        remove: (key) => {
+            return fetch(SERVICE_URL + "/" + encodeURIComponent(key), {
+                method: "DELETE",
+            });
+        },
+    });
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    // ...
+    import CustomStore from 'devextreme/data/custom_store';
+    import 'whatwg-fetch';
+
+    export default function App() {
+        const customDataSource = new CustomStore({
+            load: (loadOptions) => {
```

Comment:
Do we actually need loadOptions here?

Reply:
same for all frameworks

Reply:
We mention the **loadOptions** property in the description below. So, I think it is important to show how to use this property.

`The **load** function accepts a number of **loadOptions** (sorting, filtering, paging, etc.). Send them to a remote storage where you can generate the resulting dataset based on these properties.`